### ISSUE TITLE
Radio Button: screen readers should not select button & label separately

### ISSUE THREAD

**Issue by sulyok-daniel** [telekom/scale#1844]
**Scale Version**
3.0.0-beta.131

**Code Reproduction**
https://telekom.github.io/scale/?path=/docs/components-radio-button--standard

**Current Behavior**
When navigating with screen reader's virtual cursor the button and label parts of Scale Radio Buttons are read out separately.
This is especially an issue with Mac's Voice Over, since the label text is read twice:
![image](https://github.com/telekom/scale/assets/2097315/d36fc6d9-82a2-4de4-bb09-98166a7b039d)
![image](https://github.com/telekom/scale/assets/2097315/263a0761-bc8e-4fc9-88cd-9b6a2ea3626d)

NVDA on Win also selects the parts separately, but the label is just read once. In the screenshot above:
- button: "Anklickbar, Auswahlschalter, nicht aktiviert"
- label: "Anklickbar Radio"

**Expected Behavior**
The button and label part should be treated as an atomic unit, so a radio button option can be navigated to as a whole with screen readers to match the native radio button's behaviour: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio
